### PR TITLE
package: frontend init at 0.1.0

### PR DESCRIPTION
**Pull request by jhvst**
This PR packages the webui vite app as its own Nix derivation. For this to make sense, the app now opens with a selection screen where the user selects which repository they want to view. That repository is expected to provide static assets, which the frontend app will use.

The idea is that the frontend can be accessed on its own from the GitHub repository, which gets built in Nix and deployed to GitHub Pages. If the user chooses to pin their webui version, they can always run it by their own.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -58,6 +58,7 @@
             "teku" = inputs.ethereum-nix.packages.${system}.teku;
             # Main
             "backend" = pkgs.callPackage ./packages/backend { };
+            "frontend" = pkgs.callPackage ./webui { };
```
Comment (tupakkatapa) [inline flake.nix:61]: Can this be moved under the packages dir?

File: webui/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{ buildNpmPackage
+, nodejs
+, ...
+}:
+buildNpmPackage {
+  name = "HomestakerOS-frontend";
```
Comment (tupakkatapa) [inline webui/default.nix:6]: Lower case pls

**Review by tupakkatapa — COMMENTED** [review]
How do I run this with the backend? Why does the Vercel deployment fail?

FYI:
- Since this is going to be properly packaged, could you confirm if the FIFO file approach still works, or do we need to switch directly to using a child process?
- The flakes `devenv` script should be updated accordingly or preferably removed if it’s redundant
- `packages/backend/yarn.nix` is somewhat obsolete since it's generated from `yarn.lock` by `pkgs.mkYarnPackage`

File: flake.nix (hunk 1 above)
Comment (jhvst) [reply flake.nix:61]: Yes, but let us do that in a separate chores PR.

File: webui/default.nix (hunk 2 above)
Comment (jhvst) [reply webui/default.nix:6]: I'll let this slide, this is only visible on Nix store and when building the package. The derivation has to be updated to use a proper version later anyway.

**Pull request merged**